overlord/snapstate/snapmgr.go

```diff
@@ -509,7 +510,17 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	ss.SnapPath = downloadedSnapFile
+	// used by snapstate.Downlaod()
+	if ss.SnapPath != "" {
```

Comment:
could you invert the logic here? I find it sometimes hard to follow `if (negation) {} else {}`